### PR TITLE
Adds unit tests for backend API routes

### PR DESCRIPTION
**Pull request by hahnbi**
#### What's this PR do?
- Adds unit tests for backend API routes
- Fixes bug in prepopulation of sample test data
- Changes SolutionController.addSolution to not persist solution if user has already completed the challenge
#### What are the important parts of the code?

server/test/SpecRunner.js
#### How should this be tested by the reviewer?
1. Make sure mocha is installed globally (otherwise run `npm install mocha -g`)
2. Run `npm install` to also install the Request module for the unit tests
3. Run `mocha SpecRunner.js`, and all tests should pass
#### What are the relevant tickets? (Please add `closes`, `refs`, etc)

Refs #38 #43 #16

### CODE REVIEW

File: server/users/userController.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ userController.resetWithData = function() {
       email: null
     }).save();
   }).then(function() {
-    User.forge({
+    return User.forge({
```
Comment (alanzfu) [inline server/users/userController.js:69]: why are these returning now?

File: test/SpecRunner.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        done();
+      });
+    });
+  });
+
+  // after(function (done) {
+  //   request('http://127.0.0.1:3000/api/resetDBWithData', function () {
+  //     done();
+  //   });
+  // });
```
Comment (alanzfu) [inline test/SpecRunner.js]: is it possible to split this file up a little bit? e.g. userSpec, solutionSpec, etc.? crazy good amount of tests in here, but gonna get bloated soon.

File: server/users/userController.js (hunk 1 above)
Comment (hahnbi) [inline server/users/userController.js:69]: @alanzfu They're Promises/async functions, so they need to return in order for everything to be created in the same order consistently every time.

File: test/SpecRunner.js (hunk 2 above)
Comment (hahnbi) [inline test/SpecRunner.js]: @alanzfu Fiiiiiiiiiiiiiiiiiiine

**Comment by hahnbi** [comment]
@alanzfu split tests into separate files, ready to merge!

**Pull request merged**